test/getElementsByTagName.spec.js

```diff
@@ -0,0 +1,319 @@
+ */
+/* global it describe */
+
+import { assertResultEqual, withDom } from './utils.js'
+import { getElementsByTagName, getElementsByTagNameNS } from '../src/index.js'
+
+import tagNameLight1 from './fixtures/tagName1/light.html'
+import tagNameShadow1 from './fixtures/tagName1/shadow.html'
+
+function testTagNames (scenario, tagNames, lightDom, shadowDom, expected, expectedShadow) {
+  it(`light DOM - ${scenario}`, () => {
+    withDom(lightDom, context => {
+      assertResultEqual(tagNames, context.getElementsByTagName(tagNames), expected, true)
+    })
+  })
+
+  it(`shadow DOM - ${scenario}`, () => {
+    withDom(shadowDom, context => {
+      if (expectedShadow) {
+        expected = expectedShadow
+      }
+      assertResultEqual(tagNames, getElementsByTagName(tagNames, context), expected, true)
+    })
+  })
+}
+
+function testTagNamesNS (scenario, namespaceURI, tagNames, lightDom, shadowDom, expected, expectedShadow) {
+  it(`light DOM - ${scenario}`, () => {
+    withDom(lightDom, context => {
+      assertResultEqual(tagNames, context.getElementsByTagNameNS(namespaceURI, tagNames), expected, true)
+    })
+  })
+
+  it(`shadow DOM - ${scenario}`, () => {
+    withDom(shadowDom, context => {
+      if (expectedShadow) {
+        expected = expectedShadow
+      }
+      assertResultEqual(tagNames, getElementsByTagNameNS(namespaceURI, tagNames, context), expected, true)
+    })
+  })
+}
+
+describe('getElementsByTagName', () => {
+  const expectedSvg = [
+    {
+      tagName: 'svg',
+      classList: []
+    }
+  ]
+
+  testTagNames('handles tag names', 'svg', tagNameLight1, tagNameShadow1, expectedSvg)
+
+  const expectedSpan = [
+    {
+      tagName: 'SPAN',
+      classList: []
+    }
+  ]
+  testTagNames('handles tag names', 'span', tagNameLight1, tagNameShadow1, expectedSpan)
+
+  const expectedWildcard = [
+    {
+      tagName: 'HTML',
+      classList: []
+    },
+    {
+      tagName: 'HEAD',
+      classList: []
+    },
+    {
+      tagName: 'BODY',
+      classList: []
+    },
+    {
+      tagName: 'DIV',
+      classList: ['container']
+    },
+    {
+      tagName: 'DIV',
+      classList: ['outer-component']
+    },
+    {
+      tagName: 'DIV',
+      classList: ['wrapper']
+    },
+    {
+      tagName: 'DIV',
+      classList: ['inner-component']
+    },
+    {
+      tagName: 'SPAN',
+      classList: []
+    },
+    {
+      tagName: 'svg',
+      classList: []
+    },
+    {
+      tagName: 'defs',
+      classList: []
+    },
+    {
+      tagName: 'linearGradient',
+      classList: []
+    },
+    {
+      tagName: 'stop',
+      classList: []
+    },
+    {
+      tagName: 'stop',
+      classList: []
+    },
+    {
+      tagName: 'circle',
+      classList: []
+    }
+  ]
+
+  const expectedWildcardShadow = [
+    {
+      tagName: 'HEAD',
+      classList: []
+    },
+    {
+      tagName: 'BODY',
+      classList: []
+    },
+    {
+      tagName: 'DIV',
+      classList: ['container']
+    },
```

Comment:
Would be nice to put some capitalization variations in here to verify that that works, e.g. `DiV`, `div`, `dIv`.